static/app/components/smartSearchBar/index.tsx

```diff
@@ -864,6 +867,68 @@ class SmartSearchBar extends Component<Props, State> {
     return this.findTokensAtCursor(matchedTokens);
   }
 
+  /**
+   * Get the active filter
+   */
+  get cursorFilter() {
+    const matchedTokens = [Token.Filter] as const;
+    return this.findTokensAtCursor(matchedTokens);
+  }
+
+  get cursorValueIsoDate(): TokenResult<Token.ValueIso8601Date> | null {
+    const matchedTokens = [Token.ValueIso8601Date] as const;
+    return this.findTokensAtCursor(matchedTokens);
+  }
+
+  get cursorValueRelativeDate() {
+    const matchedTokens = [Token.ValueRelativeDate] as const;
+    return this.findTokensAtCursor(matchedTokens);
+  }
+
+  get currentFieldDefinition() {
+    if (!this.cursorToken || this.cursorToken.type !== Token.Filter) {
+      return null;
+    }
+
+    const tagName = getKeyName(this.cursorToken.key, {aggregateWithArgs: true});
+
+    const fieldDefinition = getFieldDefinition(tagName);
+
+    return fieldDefinition;
+  }
+
+  /**
+   * Determines when the date picker should be shown instead of normal dropdown options
+   * This should return true when the cursor is within a date filter and the user has
+   */
+  get shouldShowDatePicker() {
+    if (
+      !this.state.showDropdown ||
+      !this.cursorToken ||
+      this.currentFieldDefinition?.valueType !== FieldValueType.DATE ||
+      this.cursorValueRelativeDate ||
+      !(
+        this.cursorToken.type === Token.Filter &&
+        isWithinToken(this.cursorToken.value, this.cursorPosition)
+      )
+    ) {
+      return false;
+    }
+
+    const operator = this.cursorFilter?.operator ?? '';
+    const textValue = this.cursorFilter?.value?.text ?? '';
+
+    if (
+      this.cursorValueIsoDate ||
+      operator ||
+      (textValue && allOperators.find(op => op === textValue))
```

Comment:
How would you feel about, instead of exporting `allOperators` we export a function from the `searchSyntax/utils.tsx` called like `isOperator` or something?

Feels like it may be a bit more descriptive.

Reply:
I do like this better - using `isOperator()` now